### ISSUE TITLE
Undocumented behavior of number parse

### ISSUE THREAD

**Issue by 6cdh** [racket/racket#4667]
**What version of Racket are you using?**
8.9 [cs]

**What program did you run?**
I recently noticed there are some bugs in my [tree-sitter-racket](https://github.com/6cdh/tree-sitter-racket) that can parse a symbol as a number or vice versa.
I found there's some error in number grammar at [Language Model/The Reader/Reading Numbers section](https://docs.racket-lang.org/reference/reader.html#%28part._parse-number%29).

There are three types of error:

* `+i` is a (exact) number, but the grammar says it's not a number (it will be parsed as a symbol).
* `0##+i` is a (inexact) number, but the grammar says it's not a number.
* `.0@.0` is a symbol, but the grammar says it's a number.

**What should have happened?**
None

**If you got an error message, please include it here.**
None

**Please include any other relevant details**

**Comment by mflatt** [comment]
Thanks for the report!

The `.0@.0` case seems like a bug in Racket's reader. I think that should be a number, and it was a number in the pre-v7.0 reader.

The other two look like problems with the grammar.

I'll work on both the reader and the grammar.

**Comment by gus-massa** [comment]
I had too search what the `@` does. Just in case someone else is curious:

```
#lang racket
7@1               ; ==> 3.7821161410769784+5.890296893655275i
(* 7 (exp 0+1i))  ; ==> 3.7821161410769784+5.890296893655275i
```

**Comment by mflatt** [comment]
I think I fixed the grammar in 632e2ca0024bce0fa34d79f4971d2f98876dc9ac by marking some pieces near `i` as optional, but it could easily be that I'm still missing something. Closing for now, but please reopen if there are further problems.
